src/india_api/internal/service/server.py

```diff
@@ -38,6 +52,46 @@
 )
 
 
+class User(BaseModel):
+    id: str
+    email: Optional[str] = None
+
+
+class LoggedRoute(APIRoute):
+    def get_route_handler(self) -> Callable:
+        original_route_handler = super().get_route_handler()
+
+        async def custom_route_handler(request: Request) -> Response:
+            token = request.headers.get("Authorization")
+            user = None
+            log.info("token: %s", token)
+            if token:
+                token = token.split(" ")[1]
+                try:
+                    decoded_user = jwt.decode(token, options={"verify_signature": False})
+                    user_id = decoded_user.get("sub")
+                    user_email = decoded_user.get("https://openclimatefix.org/email")
+                    user = User(id=user_id, email=user_email)
+                finally:
+                    log.info("Failed to decode token")
+            # TODO: We could log referer instead of any UI flags etc.
```

Comment:
will need to use get_user_by_email - https://github.com/openclimatefix/pv-site-api/blob/main/pv_site_api/main.py#L297